src/decoder/mod.rs

```diff
@@ -893,7 +893,16 @@ impl Decoder {
                             .get_mut(&tonemap_id)
                             .ok_or(AvifError::InvalidToneMappedImage("".into()))?;
                         let mut stream = tonemap_item.stream(self.io.unwrap_mut())?;
-                        self.gainmap.metadata = mp4box::parse_tmap(&mut stream)?;
+                        let parse_tmap_res = mp4box::parse_tmap(&mut stream);
```

Comment:
nit: this inline variable is not necessary. you can simply do `match mp4box::parse_tmap(...) {`